Presto Gateway/ Expose prometheus metrics

cgupte1 opened the pull request:
1. Exposes prometheus metrics via `/prometheusMetrics` endpoint
2. Adds circleCI config for build and release
3. Adds Docker file for building `presto-gateway` image

cgupte1 committed:
Expose prometheus metrics

cgupte1 committed:
Add started circleCI configs

cgupte1 committed:
Build only current branch

cgupte1 committed:
Update image tag

cgupte1 committed:
Fix pom

cgupte1 committed:
Add custom command

cgupte1 committed:
Add docker file

cgupte1 committed:
Fix version

cgupte1 committed:
Fix job name

cgupte1 committed:
Add AWS region

cgupte1 committed:
Fix account url

cgupte1 committed:
Fix prod ecr url

cgupte1 committed:
Update docker file and circle ci

cgupte1 committed:
Fix tag

cgupte1 committed:
Remove redundant code

cgupte1 committed:
fix version

bwang-k reviewed (changes requested):
Just an overall question: Is the new Dockerfile/ CircleCI logic intended to replace everything here: https://github.com/secondmeasure/deployments/blob/master/images/presto-gateway/Dockerfile

Current logic is we manually uploaded the presto gateway jar into a "base" image so then that Dockerfile above could use it. If your new changes is intended to replace this flow, I'd recommend creating a parallel PR to remove the presto-gateway folder within images directory so we don't have redundant logic and so this image doesn't get overwritten from two places.

Alternatively, we could have this presto-gateway repo just generate the base image into the presto-gateway/base ECR repo and still keep the presto-gateway Dockerfile logic separate in deployments still. Just a thought.

cgupte1 committed:
Fix docker file version

cgupte1 commented:
@bwang-k good point. I have removed the redundant configs from deployments repo as part of https://github.com/secondmeasure/deployments/pull/423/ and updated this PR with docker compose configs.

bwang-k commented:
LGTM!
